### PR TITLE
Dev session

### PR DESCRIPTION
**Pull request by vinid**

### CODE REVIEW

File: reclist/abstractions.py [hunk 1]
```diff
@@ -126,14 +126,21 @@ def __init__(self, model: RecModel, dataset: RecDataset, y_preds: list = None):
...
         self.product_data = dataset.catalog
         self._test_results = []
         self._test_data = {}
         self._dense_repr = {}
 
-        assert len(self._y_test) == len(self._y_preds)
+        if self._y_preds is not None:
+            assert len(self._y_test) == len(self._y_preds)
+
+    def get_y_preds(self, x_test, y_test, overwrite=False):
```
Comment (vinid) [inline reclist/abstractions.py]: I am not super sure about this method and the overwrite behavior, seems order dependent and the users might get confused about this?

Comment (chloeh13q) [reply reclist/abstractions.py]: Yeah agree that it's order dependent. I think there's probably a better way of writing this. My consideration is mainly compute cost and time - we don't want to regenerate predictions for each test in the reclist, but generate new predictions as appropriate.

Comment (vinid) [reply reclist/abstractions.py]: Right, but it's not clear which is the "first" prediction set and when this should be overwritten (thinking from a user perspective).

A solution might be to have different variables for different prediction sets, in this way independently from the order in which the test runs, the results is always the same. What so you think?

Comment (patrickjohncyh) [reply reclist/abstractions.py]: We will return to this when refactoring the dataset abstraction!

**Pull request merged**